servers/xr_server.cpp

```diff
@@ -33,6 +33,16 @@
 #include "xr/xr_interface.h"
 #include "xr/xr_positional_tracker.h"
 
+XRServer::XRMode XRServer::xr_mode = XRMODE_DEFAULT;
+
+XRServer::XRMode XRServer::get_xr_mode() {
+	return xr_mode;
+}
+
+void XRServer::set_xr_mode(XRServer::XRMode p_mode) {
+	xr_mode = p_mode;
+}
+
```

Comment:
Can you also add a function that just returns whether XR is enabled. That would avoid duplicating the `switch` logic as you're currently doing in a couple of files.

Reply:
The problem is that its not on/off. There are two additional things to consider here:
1) when it is set to `XRMODE_DEFAULT` we need a value that is different in many places, when we're checking if we need to combine shaders, we check the shader setting, if we check whether OpenXR needs to be setup, its checking that value, we'll soon be adding whether OpenXR needs to be setup in the editor. So each part of the system that checks the XR mode will react differently if the mode is set to default.
2) once I start on the editor I want to switch between two ways of setting up the editor, a mode where you're purely running the editor in VR (which is the only mode available on stand alone headsets) and a mode where you still have the normal desktop UI combined with the in headset UI.

That said, at the moment I am thinking of removing the XRMODE_MIXED and just switch the scenario based on whether you're running stand alone VR or PCVR.

Reply:
I think the point was that you could add a function here that does this:
```cpp
if (get_xr_mode() == XRServer::XRMODE_DEFAULT) {
	return GLOBAL_GET("xr/shaders/enabled");
} else {
	return get_xr_mode() == XRServer::XRMODE_ON;
}
```

Reply:
@akien-mga well the original idea was that it was off/on/mixed but I've removed the mixed and solved it exactly as you suggested :)

Reply:
No my point is that `RenderingCompositor` and `OpenXRAPI` shouldn't have to duplicate this check. They should just call `XRServer::is_xr_enabled()` which would do what the above snippet does.

Reply:
@akien-mga but that doesn't work because when XRMODE_DEFAULT is set, they check different switches.

OpenXRAPI checks whether OpenXR needs to be enabled (hate the need to have this one but seeing OpenXR needs to take ownership of Vulkan we are living with this PITA setting)
RenderingCompositor checks whether the XR shader setting is enabled

And in the near future the editor will check an editor specific setting that tells us if the editor should run in XR mode and whether OpenXR needs to be enabled in the editor.

Setting XRMODE_ON overrides all these settings and enables them all. XRMODE_OFF turns them all off.

Hence the need for this check to be implemented **differently** in all this situations.

Reply:
The XRMODE_OFF use case is just to shut everything off, so people can create an app that they setup with the proper settings to use it for XR, but then can shut everything off to also start it in desktop mode.
The XRMODE_ON use case is mostly the editor, as it always turns on OpenXR you cant really use it if you're using a different SDK like ARKit or ARCore.